### PR TITLE
11347 supporting azurerm subscription tags

### PR DESCRIPTION
**Pull request by crkg**
@jackofallops this is reference to - https://github.com/terraform-providers/terraform-provider-azurerm/pull/11347 conversation to add tags to azurerm subscription resource. please approve the workflow.

### CODE REVIEW

**Comment by crkg** [comment]
@jackofallops , can you please approve the workflow. we will close this topic.

File: azurerm/internal/services/subscription/subscription_resource.go
Comment (jackofallops) [inline azurerm/internal/services/subscription/subscription_resource.go:445]: We shouldn't need to do anything with the tags here, as we will be disabling the subscription.
```suggestion
```

File: azurerm/internal/services/subscription/subscription_resource.go
Comment (jackofallops) [inline azurerm/internal/services/subscription/subscription_resource.go:465]: We only ever return nil at the end of a delete, returning read here will result in Terraform presenting an erroneous not-found error to the user on success.
```suggestion
	return nil
```

File: azurerm/internal/services/subscription/subscription_resource_test.go
Comment (jackofallops) [inline azurerm/internal/services/subscription/subscription_resource_test.go:71]: Can we keep these in long-hand form as with the rest of the test suite?

File: azurerm/internal/services/subscription/subscription_resource_test.go
Comment (jackofallops) [inline azurerm/internal/services/subscription/subscription_resource_test.go:123]: This shouldn't be necessary. The exists and implicit read/import functionality will check the values for the tags.  The `Exists` function itself is simply to check the resource is present, not for any particular setting / configuration
```suggestion
```

File: azurerm/internal/services/subscription/subscription_resource_test.go
Comment (jackofallops) [inline azurerm/internal/services/subscription/subscription_resource_test.go:191]: This is not supported here and will create a test config error when run
```suggestion
```

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @crkg - Thanks for this PR, it's off to a good start. I've left some comments and changes in line below. If you can address those, I'll re-review asap. 
Thanks!

**Comment by crkg** [comment]
Thanks Steve, i will work on the recommendation and push the changes.

**Comment by crkg** [comment]
> Hi @crkg - Thanks for this PR, it's off to a good start. I've left some comments and changes in line below. If you can address those, I'll re-review asap.
> Thanks!

Hi Steve, 

PR is ready to be merged with no conflicts. please take a moment to review and merge. 

Thank you.

File: azurerm/internal/services/subscription/subscription_resource.go
Comment (jackofallops) [inline azurerm/internal/services/subscription/subscription_resource.go:289]: We can remove this, it should be handled by the `Read`.
```suggestion
```

File: azurerm/internal/services/subscription/subscription_resource_test.go
Comment (jackofallops) [reply azurerm/internal/services/subscription/subscription_resource_test.go:91]: Looks like this one was missed from the last review? Can we get these in the form:
```suggestion
				check.That(data.ResourceName).ExistsInAzure(r),
				check.That(data.ResourceName).Key("tags.%").HasValue("2"),
				check.That(data.ResourceName).Key("tags.cost_center").HasValue("MSFT"),
				check.That(data.ResourceName).Key("tags.environment").HasValue("Production"),
```
The consistency helps us make bulk changes if/when we update approaches to the test framework.

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @crkg - Thanks for the updates, I've left a couple more comments inline below. If you can address those I'll make time to test this as the Subscription resource isn't compatible with our CI test runner currently, so needs running by hand.

Thanks!

**Comment by gro1m** [comment]
Hey @crkg it looks like you missed a review for your test file. I saw this when looking at the file view: https://github.com/terraform-providers/terraform-provider-azurerm/pull/12287/files and for some reason the Terraform Schema Linting seems to have failed - just in case you did not notice.

**Comment by jackofallops** [comment]
Hi @crkg - There's still an outstanding change requested from the previous review, specifically updating the test cases where `assert := check.That(data.ResourceName)` is used to being the long-hand version.  If you can update those, I'll re-review and get the tests run. 👍 
Thanks!

**Comment by crkg** [comment]
> Hi @crkg - There's still an outstanding change requested from the previous review, specifically updating the test cases where `assert := check.That(data.ResourceName)` is used to being the long-hand version. If you can update those, I'll re-review and get the tests run. 👍
> Thanks!

Hi Steve, sorry i missed to commit those changes in previous commit. thank you for the suggestions.

**Comment by crkg** [comment]
> > Hi @crkg - There's still an outstanding change requested from the previous review, specifically updating the test cases where `assert := check.That(data.ResourceName)` is used to being the long-hand version. If you can update those, I'll re-review and get the tests run. 👍
> > Thanks!
> 
> Hi Steve, sorry i missed to commit those changes in previous commit. thank you for the suggestions. its fixed please approve the workflow.

@jackofallops ( Steve) can you please review and approve the changes.

**Comment by jackofallops** [comment]
Hi @crkg - We have an issue testing subscription creation/update currently, so I'm afraid this is blocked until that support ticket is resolved. I'll loop back to this just as soon as we're able to test.

**Comment by crkg** [comment]
> Hi @crkg - We have an issue testing subscription creation/update currently, so I'm afraid this is blocked until that support ticket is resolved. I'll loop back to this just as soon as we're able to test.

Hi Steve, thank you. let me know when we are good to loop back this PR.

**Comment by crkg** [comment]
> > Hi @crkg - We have an issue testing subscription creation/update currently, so I'm afraid this is blocked until that support ticket is resolved. I'll loop back to this just as soon as we're able to test.
> 
> Hi Steve, thank you. let me know when we are good to loop back this PR.

Hi Steve, may i know if this feature can be plugged back, I was waiting for a confirmation on a blocked support ticket if its resolved?

**Comment by donwlewis** [comment]
It would be great to see this get implemented soon.  Thanks!

**Comment by pinophyta** [comment]
Looking forward to having this useful feature in the provider soon!

**Comment by katbyte** [comment]
Hey all, sorry that this PR has been languishing here for so long. We're still working to unblock our ability to test - but progress has finally been made on that front so are now working towards getting this queued up. No timeframe to share yet but hopefully sooner than later!

**Comment by Eslam10** [comment]
Hello, 
Would someone please share the updates on this ? 
Thank you.

**Comment by jackofallops** [comment]
Hi @crkg - we've been able to resolve the testing issues for this. A lot has changed in the provider structure since this was first opened, so I've had to take your work forward to another PR to add changes I found to be necessary. Thanks again for your work on this, and your continued patience in getting this through.  I'm going to close this PR out, please subscribe to #14445 for further updates.

**Comment by crkg** [comment]
> Hi @crkg - we've been able to resolve the testing issues for this. A lot has changed in the provider structure since this was first opened, so I've had to take your work forward to another PR to add changes I found to be necessary. Thanks again for your work on this, and your continued patience in getting this through. I'm going to close this PR out, please subscribe to #14445 for further updates.

Thank you for the feature, much appreciated.

**Pull request closed without merging**
